### PR TITLE
fix(components): [el-select] fix not clear hoverIndex

### LINKED ISSUE #4294: [Bug Report] Hover index of el-select is still there after clearing input
### Element Plus version
1.2.0-beta.1

### OS/Browsers version
Windows 10 / Chrome 95.0.4638.69

### Vue version
3.2.21

### Reproduction Link
https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtc2VsZWN0IHYtbW9kZWw9XCJ2YWx1ZVwiIGNsZWFyYWJsZSBtdWx0aXBsZSBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiPlxuICAgIDxlbC1vcHRpb25cbiAgICAgIHYtZm9yPVwiaXRlbSBpbiBvcHRpb25zXCJcbiAgICAgIDprZXk9XCJpdGVtLnZhbHVlXCJcbiAgICAgIDpsYWJlbD1cIml0ZW0ubGFiZWxcIlxuICAgICAgOnZhbHVlPVwiaXRlbS52YWx1ZVwiXG4gICAgLz5cbiAgPC9lbC1zZWxlY3Q+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZSA9IHJlZignJylcbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjEnLFxuICAgIGxhYmVsOiAnT3B0aW9uMScsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjInLFxuICAgIGxhYmVsOiAnT3B0aW9uMicsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjMnLFxuICAgIGxhYmVsOiAnT3B0aW9uMycsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjQnLFxuICAgIGxhYmVsOiAnT3B0aW9uNCcsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjUnLFxuICAgIGxhYmVsOiAnT3B0aW9uNScsXG4gIH0sXG5dXG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==

### Steps to reproduce
1. Click on el-select to open all options
2. Select 'Option3', 'Option4' and 'Option5' by clicking on these 3 options
3. Click outside of el-select to close all options
4. Clear all selected options by clicking on the little cross in the input suffix
5. Click on el-select again to open all options
=> Hover index effect (gray background color) is on 'Option3'

### What is Expected?
After clearing all selected options the hover index effect (gray background color) should be not visible.

### What is actually happening?
The hover index effect is still visible after clearing all selected options and is set to an option I pressed some clicks ago.

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #4294

### CODE REVIEW

File: packages/components/select/src/useSelect.ts [hunk 1]
```diff
@@ -569,6 +571,7 @@ export const useSelect = (props, states: States, ctx) => {
     ctx.emit(UPDATE_MODEL_EVENT, value)
     emitChange(value)
     states.visible = false
+    states.hoverIndex = -1
```
Comment (msidolphin) [inline packages/components/select/src/useSelect.ts:574]: Why set hoverIndex to -1 when delete selected

Comment (msidolphin) [reply packages/components/select/src/useSelect.ts:574]: I think we should set the value of hoverIndex like virtual select

Comment (msidolphin) [reply packages/components/select/src/useSelect.ts:574]: Now the hoverIndex does not change when mouseover

Comment (btea) [reply packages/components/select/src/useSelect.ts:574]: > Why set hoverIndex to -1 when delete selected?

`hoverIndex`  indicates the subscript of the option of the hover state. Therefore, when the hover state is cleared, its value is set to -1.

Comment (btea) [reply packages/components/select/src/useSelect.ts:574]: > Now the hoverIndex does not change when mouseover  

This does not seem to be caused by this change. The current performance of the official website is the same. The selected options have no hover effect.

**Pull request closed without merging**